### PR TITLE
fix: Slider step prop not accepting null value

### PR DESCRIPTION
**Pull request by shangyilim**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

1. [#26976](https://github.com/ant-design/ant-design/issues/26976)

### 💡 Background and solution

The `step` prop for `Slider` will trigger typescript error: `Type null is not assignable to type 'number | undefined' `.  This is because the type definition does not allow `null` value, eventhough in the [documents it allowed as mentioned](https://ant.design/components/slider/):

```
the granularity the slider can step through values. Must greater than 0, and be divided by (max - min) . When marks no null, step can be null
```

This can be fixed by changing the definition of `step` to allow `null` value. 
Since underlying component used in `Slider` is `react-component/slider` its `step` props do not allow `null`, although in their documentation [it is allowed.](https://github.com/react-component/slider).
A quick fix will force the value `null!` within the library, while client apps will still be able to pass in `step={null}` without typescript errors. Until the underlying prop interface is resolved, this should a sufficient enough fix.

Updated typescript test to catch this error, and also added tests to ensure slider behaves as expected when null is passed. 
### 📝 Changelog

No breaking changes

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |Fixed `step` prop in `Slider` to accept null as a valid value in accordance to the documentation.|
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/slider/index.tsx [hunk 1]
```diff
@@ -159,6 +160,7 @@ const Slider = React.forwardRef<unknown, SliderSingleProps | SliderRangeProps>(
     return (
       <RcSlider
         {...(restProps as SliderSingleProps)}
+        step={restProps.step!}
```
Comment (afc163) [inline components/slider/index.tsx:163]: RcSlider 的定义也需要修正一下，避免用 `!`。

**Comment by afc163** [comment]
image

Could you fill the changelog part?

File: components/slider/index.tsx (hunk 1 above)
Comment (shangyilim) [reply components/slider/index.tsx:163]: Unfortunately until RcSlider fixes their definition, there is nothing we can do on our end and to avoid typescript lint errors.
Furthermore I have added another test to safeguard againts `undefined`:
```
it('step should not crash with undefined value', () => {
    [undefined, null].forEach(value => {
      mount(<Slider step={value} tooltipVisible />);
    });
  });
```
